src/core/Akka.Tests/Actor/GracefulStopSpecs.cs

```diff
@@ -0,0 +1,71 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="GracefulStopSpecs.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Actor.Dsl;
+using Akka.TestKit;
+using Akka.TestKit.TestActors;
+using Xunit;
+using FluentAssertions;
+
+namespace Akka.Tests.Actor
+{
+    public class GracefulStopSpecs : AkkaSpec
+    {
+        [Fact(DisplayName = "GracefulStop should terminate target actor on-time")]
+        public async Task GracefulStopShouldTerminateOnTime()
+        {
+            // arrange
+            var actor = Sys.ActorOf(BlackHoleActor.Props);
+            Watch(actor);
+
+            // act
+            var stopped = await actor.GracefulStop(TimeSpan.FromSeconds(3));
+            await ExpectTerminatedAsync(actor);
+
+            // assert
+            stopped.Should().BeTrue();
+        }
+
+        [Fact(DisplayName = "GracefulStop should return true for an already terminated actor")]
+        public async Task GracefulStopShouldReturnTrueForAlreadyDeadActor()
+        {
+            // arrange
+            var actor = Sys.ActorOf(BlackHoleActor.Props);
+            Watch(actor);
+
+            // act
+            Sys.Stop(actor);
+            await ExpectTerminatedAsync(actor);
+            var stopped = await actor.GracefulStop(TimeSpan.FromSeconds(3));
+
+            // assert
+            stopped.Should().BeTrue();
+        }
+
+        private class CustomShutdown{}
+
+        [Fact(DisplayName = "GracefulStop should return false if shutdown goes overtime", Skip = "GracefulStop currently throws a TaskCancellationException, which seems wrong")]
```

Comment:
Marked this test as `Skipped` for now - going to change that in https://github.com/akkadotnet/akka.net/pull/6102

This API is wrong. We shouldn't throw an `Exception` if graceful stop fails to complete on-time - we should simply return `false`. Otherwise there's no point to having a return value at all.